app/submodule/wallet/remotewallet/keymap.go

```diff
@@ -20,10 +20,16 @@ func GetKeyType(p address.Protocol) core.KeyType {
 }
 
 func ConvertRemoteKeyInfo(key *crypto.KeyInfo) *core.KeyInfo {
-	return &core.KeyInfo{
-		PrivateKey: key.Key(),
-		Type:       core.SignType2Key(key.SigType),
-	}
+	var ki *core.KeyInfo
+
+	_ = key.UsePrivateKey(func(privateKey []byte) error {
+		ki = &core.KeyInfo{
+			Type:       core.SignType2Key(key.SigType),
+			PrivateKey: privateKey,
```

Comment:
You won't be able to use the parameter of the closure in the outside context, since it has been destroyed after `UsePrivateKey` returns. Accessing it should result in a panic. It is possible that this is why the unit tests fails. So in cases like this, where you have to interface with code outside your control, you likely _have_ to copy into an insecure buffer. There are a few more places where `UsePrivateKey` is used like this, the same applies over there.

Reply:
Thanks for your comment. Correction done